infrastructure/create_secret

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+
+function usage() {
+  echo "Usage: ./$0 [variable-name]"
+  echo "  eg: ./$0 lobby_db_password"
+  echo ""
+  echo "This script will encode a secret value that can be embedded in ansible"
+  echo "and decrypted when ansible runs given the vault password"
+  echo ""
+  echo "This script expects the secret value to be encrypted to be written" 
+  echo "to a file called 'secret'"
+  echo "This script expects a second file to contain the ansible vault password"
+  echo "named 'vault_password'"
+  exit 1
+}
+
+variable_name=$1
+
+set -eu
+
+if [ ! -f "secret" ]; then
```

Comment:
Hmm, not sure if I like the pattern of relying on some specific files to exist.
In my opinion we should be able to pass a filename and/or fall back to a default value.

Reply:
The files are temporary and are added to .gitignore to avoid accidental checkin. The file is there so you don't type the command on prompt (avoid accidental typo) nor have to add the secret via command line arg (insecure).

Using 'create_secret' is one time, the vault password file and secret file should be removed afterwards. See the README update for the command that is being replaced, this is really a wrapper around a one-liner.

Let me know if that makes sense.

Reply:
Are all files temporary?
If they are, why not create files in `/tmp` and reference them in other parts of the script?
This way there's no chance it can be checked in accidentally.

Either way, the current system "enforces" some arbitrary filenames, that might conflict with pre-existing files for some reason (when used). The typo argument is valid, but at least I'm personally using tab-autcompletion for files to prevent exactly that.
I just think it's more natural to be explicit about the files your specifying this as input.
To me this feels like if grep would only accept input files called "input" in the current directory, and you'd have to copy files or use symlinks to run it on any other file.

Different thought: What about using STDIN for the secret? Avoids the problem with the secret being stored in bash history, in if you're really paranoid you can also use `read -s SECRET` to read from the terminal in a hidden way and store them into `$SECRET`.

Reply:
> Are all files temporary?

Mostly temporary, we should not be keeping secrets, either the vault password or unencrypted secret files on our systems for very long. Those values represent a security concern/vulnerability and should be deleted in a reasonable time frame.

> If they are, why not create files in /tmp and reference them in other parts of the script?

Being in `/tmp` might help them be cleaned up, but probably better for it to be explicit as we should know what we're doing when dealing with secrets. 

The vault-password file is used by ansible-playbook commands proper, having it be in a known location is preferable. 

Second, '/tmp' tends to have more lenient read permissions, isn't it a bit more locked down to have it in a folder owned by only the current user?

Re: typo -- I'm concerned about typo's using STDIN. 

> I just think it's more natural to be explicit about the files your specifying this as input.

The issue I see with this is then parameter ordering becomes extremely significant. If we go the args flags, I seriously raise the prospect we are over-engineering this.

> To me this feels like if grep would only accept input files called "input" in the current directory, and you'd have to copy files or use symlinks to run it on any other file.

The idea is you encrypt one secret at a time. Update 'secret' with the secret, pass the parameter name as a variable to this script, copy the output into a '.yml' file, repeate until you encrypted all the variables you want, then delete the 'secrets' file. Once done testing, delete also the 'vault-password' file.

Reply:
Ok, I get why tmp files are maybe not the right way to go. Tmp has special permissions so that only the owner of a file can actually delete it, I guess if adjusting the read permission, we probably could keep the file private, blocked for other users. But this is just my assumption at this point.

I personally think that the read command is very intriguing; there's no need for a magic file to exist, and if you're still concerned about potential typos, you could still pipe a magic file into the script so it stays the same.

Let me know what you think.
If you disagree, feel free to merge but I'm convinced this is definitely an option worth considering